.github/workflows/ci.yml

```
@@ -0,0 +1,167 @@
+        
+    - name: Run tests with pytest
+      run: |
+        pytest -v --tb=short
+        
+    - name: Run tests with coverage
+      run: |
+        pip install pytest-cov
+        pytest --cov=. --cov-report=xml --cov-report=term-missing --cov-fail-under=80
+        
+    - name: Upload coverage reports
+      uses: codecov/codecov-action@v3
```

actionlint (1.7.7)

the runner of "codecov/codecov-action@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)